codalab/client/json_api_client.py

```diff
@@ -78,13 +78,21 @@ def __init__(self, type_, id_, *args):
         self.setdefault('id', id_)
 
     def as_linkage(self):
-        """Serialize into relationship linkage dict for JSON API requests."""
-        return {
-            'data': {
-                'type': self.type_,
-                'id': self.id_,
+        """
+        Serialize into relationship linkage dict for JSON API requests.
+        Empty relationships should be properly serialized as a null linkage.
+        """
+        if self:
```

Comment:
`self` is falsey if it's an empty relationship, as defined in JsonApiClient.__nonzero__. This might be clearer if I revert back to having a separate subclass for empty relationships.